apps/admin-portal/src/api/application.ts

```diff
@@ -211,3 +211,144 @@ export const getApplicationList = (limit: number, offset: number): Promise<any>
             return Promise.reject(error);
         });
 };
+
+/**
+ * Gets the available inbound protocols.
+ *
+ * @param customOnly If true only returns custom protocols.
+ */
+export const getAvailableInboundProtocols = (customOnly: boolean): Promise<any> => {
+    const requestConfig = {
+        headers: {
+            "Accept": "application/json",
+            "Access-Control-Allow-Origin": GlobalConfig.clientHost,
+            "Content-Type": "application/json"
+        },
+        method: HttpMethods.GET,
+        url: ServiceResourcesEndpoint.applications + "/meta/inbound-protocols?customOnly=" + customOnly
+    };
+
+    return httpClient.request(requestConfig)
+        .then((response) => {
+            if (response.status !== 200) {
+                return Promise.reject(new Error("Failed get Inbound protocols from: "));
+            }
+            return Promise.resolve(response.data as AuthProtocolMetadataInterface);
+        }).catch((error) => {
+            return Promise.reject(error);
+        });
+};
+
+/**
+ * Gets the OIDC protocol's meta data.
+ *
+ * @return {Promise<any>} A promise containing the response.
+ */
+export const getOIDCMetadata = (): Promise<any> => {
+    const requestConfig = {
+        headers: {
+            "Accept": "application/json",
+            "Access-Control-Allow-Origin": GlobalConfig.clientHost,
+            "Content-Type": "application/json"
+        },
+        method: HttpMethods.GET,
+        url: ServiceResourcesEndpoint.applications + "/meta/inbound-protocols/oidc"
+    };
+
+    return httpClient.request(requestConfig)
+        .then((response) => {
+            if (response.status === 404) {
+                return Promise.reject(new Error("Inbound protocol not configured"));
+            } else if (response.status !== 200) {
+                return Promise.reject(new Error("Failed get OIDC meta data from: "));
+            }
+            return Promise.resolve(response.data as OIDCMetadataInterface);
+        }).catch((error) => {
+            return Promise.reject(error);
+        });
+};
+
+/**
+ * Gets the application's OIDC data.
+ *
+ * @param id Application ID
+ */
+export const getOIDCData = (id: string): Promise<any> => {
+    const requestConfig = {
+        headers: {
+            "Accept": "application/json",
+            "Access-Control-Allow-Origin": GlobalConfig.clientHost,
+            "Content-Type": "application/json"
+        },
+        method: HttpMethods.GET,
+        url: ServiceResourcesEndpoint.applications + "/" + id + "/inbound-protocols/oidc"
+    };
+
+    return httpClient.request(requestConfig)
+        .then((response) => {
+            if (response.status !== 200) {
```

Comment:
Let's remove the if condition here as well.

Reply:
same as mentioned above